src/components/Input.tsx

```diff
@@ -18,20 +21,43 @@ export default function Input({
   value,
   message,
   autoFocus = true,
+  maxLength,
   ...props
 }: InputPropsType) {
+  const setClassNameByProperty = (property: string) => {
+    if (property === 'error') return 'border-b-sub-1'
+    if (property === 'success') return 'border-b-primary-1'
+
+    return 'border-b-grey-4'
+  }
+
+  const setMessageColorByProperty = (property: string) => {
+    return property === 'success' ? 'text-primary-2' : 'text-sub-1'
+  }
+
   return (
     <div>
-      <p className="pb-6">{label}</p>
-      <input
-        name={name}
-        value={value}
-        placeholder={placeholder}
-        className={`w-full border-b border-b-grey-4 pt-4 pb-4 placeholder:text-sm`}
-        autoFocus={autoFocus}
-        {...props}
-      />
-      <p className="pt-3 text-sm text-primary-2">{message}</p>
+      {label && <p className="pb-6">{label}</p>}
+      <div className="relative flex items-center">
+        <input
+          type="search"
+          name={name}
+          value={value}
+          placeholder={placeholder}
+          autoFocus={autoFocus}
+          maxLength={maxLength}
+          className={`w-full border-b border-solid pt-4 pb-4 placeholder:text-sm 
+          focus:outline-none ${setClassNameByProperty(property)}
+        `}
+          {...props}
+        />
+        <CloseIcon className="absolute right-0 hover:cursor-pointer" />
+      </div>
+      {property !== 'default' && (
+        <p className={`pt-3 text-sm ${setMessageColorByProperty(property)}`}>
+          {message}
```

Comment:
setMessageColorByProperty함수를 가독성떄문에 이렇게 만드신거같은데 이처럼 간단하게 삼항연산자로 해결하는건 저는 함수로 안빼도 될거같아요.
함수로 뺀다는거는 중복사용되서 코드를 줄이기 위함이거나 추상화를 위해서 함수분리할떄 한다고 생각하는데 여기는 둘다 해당하지 않는거같아서영

Reply:
원래 따로 빼지 않았었는데 메시지 하나만 보여주는 코드임에도 불구하고 코드가 조금 길어진다고 생각해서 따로 뺐습니다!
저 부분을 함수로 빼지 않으면 

```suggestion
      {property !== 'default' && (
        <p className={`pt-3 text-sm  ${
          property === 'success' ? 'text-primary-2' : 'text-sub-1'}`}>
          {message}
        </p>
      )}
```

이렇게 보이는데  그래도 분리하지 않는것이 좋을까요?

Reply:
음 저는 이정도는 코드 가독성에는 문제 안된다고 생각합니다!
근데 앞으로 제가 말했던 위의 원리에 부합한다면 꼭 함수로 뺴면 좋을거같아용!